app/models/delayed/job.rb

```diff
@@ -11,25 +11,25 @@ class Job < ::ActiveRecord::Base
     # high-level queue states
     scope :live, -> { where(failed_at: nil) }
     scope :failed, -> { where.not(failed_at: nil) }
-    scope :erroring, -> { where.not(last_error: nil) }
+    scope :erroring, -> { where.not(last_error: nil).merge(unscoped.live) }
 
     # claim/lock states
-    scope :claimed, -> { where.not(locked_at: nil) }
-    scope :claimed_by, ->(worker) { where(locked_by: worker.name) }
-    scope :unclaimed, -> { where(locked_at: nil) }
+    scope :claimed, -> { where.not(locked_at: nil).merge(unscoped.live) }
+    scope :claimed_by, ->(worker) { where(locked_by: worker.name).merge(unscoped.live) }
+    scope :unclaimed, -> { where(locked_at: nil).merge(unscoped.live) }
 
     # run_at states
-    scope :future, ->(as_of: db_time_now) { where(arel_table[:run_at].gt(as_of)) }
-    scope :pending, ->(as_of: db_time_now) { where(arel_table[:run_at].lteq(as_of)) }
+    scope :future, ->(as_of: db_time_now) { merge(unscoped.live).where(arel_table[:run_at].gt(as_of)) }
+    scope :pending, ->(as_of: db_time_now) { merge(unscoped.live).where(arel_table[:run_at].lteq(as_of)) }
 
     # workability states
-    scope :working, -> { claimed.live }
-    scope :workable, -> { unclaimed.live.pending }
+    scope :working, -> { claimed }
+    scope :workable, -> { unclaimed.pending }
     scope :workable_by, ->(worker) {
       pending
-        .merge(unclaimed.or(where(arel_table[:locked_at].lt(db_time_now - lock_timeout))))
+        .merge(where(locked_at: nil).or(where(arel_table[:locked_at].lt(db_time_now - lock_timeout))))
```

Comment:
I have a follow-up PR that simplifies this, as part of making the monitor's `working_count` obey lock timeouts.

But for now I opted to avoid changing the pickup query to avoid overcomplicating this PR -- as-is, tests prove that the pickup query has not changed.